pkg/external-provider/errors.go

```diff
@@ -0,0 +1,21 @@
+package provider
+
+import "errors"
+
+var (
+	// ErrNewOperatorNotSupportedByPrometheus creates an error that represents the fact that we were requested to service a query that
+	// Prometheus would be unable to support.
+	ErrNewOperatorNotSupportedByPrometheus = errors.New("operator not supported by prometheus")
```

Comment:
nit: let's just call this `ErrNotSupportedByPrometheus`

Reply:
(along with the error message `not supported by prometheus`)

Reply:
just a suggestion: doesn't `ErrUnsupportedQuery` make more sense here?

Reply:
I think `ErrUnsupportedOperator` going by the error text. Opinion?

Reply:
the more explicit the better init